include/lsst/afw/geom/SpherePoint.h

```diff
@@ -0,0 +1,343 @@
+// -*- LSST-C++ -*-
+
+/*
+ * LSST Data Management System
+ * See COPYRIGHT file at the top of the source tree.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program. If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_SPHEREPOINT_H_
+#define LSST_AFW_GEOM_SPHEREPOINT_H_
+
+#include <ostream>
+#include <utility>
+
+#include "lsst/afw/geom/Angle.h"
+#include "lsst/afw/geom/Point.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+    /**
+     * @brief Point in an unspecified spherical coordinate system.
```

Comment:
I find the "unit sphere" wording confusing (we have a representation as unit vectors, sure, but you can use the class exclusively through longitude/latitude if you want). I can add an explicit "does not have distance to an object" to the first paragraph, though.